frontend/test/metabase/lib/expressions/typechecker.unit.spec.js

```diff
@@ -0,0 +1,136 @@
+import { parse } from "metabase/lib/expressions/parser";
```

Comment:
Maybe it's worth including a comment here like the one in the PR comment.

> Since the type checking is inserted as the last stage in the expression parser, the whole tests must continue to pass (i.e. none of them should thrown an exception) to assert that type checker works correctly.

Without that context, it's a bit weird that the code we're testing isn't directly imported.

Reply:
Added!